pkg/cidata/cidata_test.go

```diff
@@ -0,0 +1,56 @@
+package cidata
+
+import (
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/lima-vm/lima/pkg/limayaml"
+	qemu "github.com/lima-vm/lima/pkg/qemu/const"
+	"github.com/xorcare/pointer"
+	"gotest.tools/v3/assert"
+)
+
+func TestSetupEnv(t *testing.T) {
+	urlMustParse := func(urlStr string) *url.URL {
+		u, err := url.Parse(urlStr)
+		if err != nil {
+			panic(err)
+		}
+
+		return u
+	}
+
+	httpProxyCases := []*url.URL{
+		urlMustParse("http://127.0.0.1:8080"),
+		urlMustParse("https://127.0.0.1:8080"),
+		urlMustParse("sock4://127.0.0.1:8080"),
+		urlMustParse("sock5://127.0.0.1:8080"),
+		urlMustParse("http://127.0.0.1:8080/"),
+		urlMustParse("http://127.0.0.1:8080/path"),
+		urlMustParse("http://localhost:8080"),
+		urlMustParse("http://localhost:8080/"),
+		urlMustParse("http://localhost:8080/path"),
+		urlMustParse("http://docker.for.mac.localhost:8080"),
+		urlMustParse("http://docker.for.mac.localhost:8080/"),
+		urlMustParse("http://docker.for.mac.localhost:8080/path"),
```

Comment:
`docker.for.mac.localhost` must be in your hosts `/etc/hosts` file; it does not resolve otherwise:

```
--- FAIL: TestSetupEnv (0.13s)
    --- FAIL: TestSetupEnv/docker.for.mac.localhost:8080 (0.00s)
        cidata_test.go:45: assertion failed: http://docker.for.mac.localhost:8080 (envs[envKey] string) != http://192.168.5.2:8080 (string)
    --- FAIL: TestSetupEnv/docker.for.mac.localhost:8080#01 (0.00s)
        cidata_test.go:45: assertion failed: http://docker.for.mac.localhost:8080/ (envs[envKey] string) != http://192.168.5.2:8080/ (string)
    --- FAIL: TestSetupEnv/docker.for.mac.localhost:8080#02 (0.00s)
        cidata_test.go:45: assertion failed: http://docker.for.mac.localhost:8080/path (envs[envKey] string) != http://192.168.5.2:8080/path (string)
```

Reply:
use a fakeLookupIP to resolved